packages/webapp/pages/onboarding.tsx

```diff
@@ -140,6 +148,25 @@ export function OnboardPage(): ReactElement {
     }
   }, [isSeniorUser, isFeedSettingsDefined, updateAdvancedSettings]);
 
+  useEffect(() => {
+    if (!isPageReady || isLogged.current) {
+      return;
+    }
+
+    if (user) {
+      router.replace(getPathnameWithQuery(webappUrl, window.location.search));
+      return;
+    }
+
+    isLogged.current = true;
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isPageReady, user]);
+
+  useEffect(() => {
+    setHasSelectTopics(!!feedSettings?.includeTags?.length);
+  }, [feedSettings?.includeTags?.length]);
```

Comment:
IMO if there are no other uses of this `setHasSelectTopics` we can remove both `useState` and `useEffect` and just writing:

```ts
const hasSelectTopics = !!feedSettings?.includeTags?.length;
```

when there are no huge calculation is even better not using useMemo for those kind of vars. What do you think?

Reply:
@nensidosari not sure if you did this with a specific edge-case or not?

Reply:
Can't really remember, and at the moment seems only to be used in loggign event skip tag selection (which if u remember u discover it still fires, we just don't know how/why 😅 ). I think we're good to do what Luca says

Reply:
I think this is some legacy thing where we allowed skipping.

Reply:
Worst case to support, we can convert this to a ref since we only use it to check some values within a function, not during render phase.

Reply:
Yeah, either a usememo or use ref.